programming-extensions/programming-extension-vfsprovider/src/main/java/org/objectweb/proactive/extensions/dataspaces/vfs/VFSFactory.java

```diff
@@ -244,8 +271,25 @@ private static FilesCache getCacheType() {
         return cache;
     }
 
-    private static FileSystemOptions createDefaultFileSystemOptions() throws FileSystemException {
+    private static synchronized FileSystemOptions createDefaultFileSystemOptions(UserCredentials credentials)
+            throws FileSystemException {
+
         final FileSystemOptions options = new FileSystemOptions();
+        if (logger.isTraceEnabled()) {
+            logger.trace("CREDENTIALS = " + credentials);
+        }
+        if (credentials != null && !credentials.isEmpty()) {
+            UserAuthenticator auth = new StaticUserAuthenticator(credentials.getDomain(),
```

Comment:
I do not see a UserAuthenticator with the private key (instead of the password). Would you explain that please

Reply:
You are right, my mistake, I need to implement it.